### ISSUE TITLE
steppedOn event on block registries cannot handle blocks that are not full blocks

### ISSUE THREAD

**Issue by HaloSense** [KubeJS-Mods/KubeJS#875]
### Minecraft Version

1.20.1

### KubeJS Version

2001.6.5-build.7

### Rhino Version

2001.2.2-build.18

### Architectury Version

9.2.14

### Forge/Fabric Version

47.2.32

### Describe your issue

I am trying to register a block that is not a full block (i.e. doesn't have the hitbox of a full block, such as vanilla carpet) and design some `steppedOn()` effect for it. But I find out that `steppedOn()` event doesn't look for entities stepping on this block itself (given the block is not full, the coordinates of the said block is the same as the entity). Rather, it looks for entities stepping on the block 1 block above it (i.e. the entity's y coordinate is higher than the block by 1). Is it possible to add an option for `steppedOn()` event to have different trigger logics for both full blocks and non-full blocks?

**Comment by HaloSense** [comment]
> I encountered the same issue, and my tests found that when the y of the box is greater than or equal to around 5, steppedOn() works normally. This may not be a problem with KubeJS; it could be the original mechanism

Yes, I remember having tested some more conditions after posting this issue, and the mechanism works fine with blocks larger than or equal to half a block high. I wonder whether it actually is the designated outcome.

**Comment by FutureLin-JM** [comment]
I’ve been working on a Java mod recently and remembered this issue, so I did some testing. The `steppedOn()` method uses the `stepOn` method in Java, and its testing results are basically consistent with KubeJS.
If you want to achieve the carpet thickness effect, you should use the `entityInside` method. Unfortunately, this method was not introduced in KubeJS 1.20.1.
